train.py

```diff
@@ -173,8 +180,19 @@
         if k.startswith(unwanted_prefix):
             state_dict[k[len(unwanted_prefix) :]] = state_dict.pop(k)
     model.load_state_dict(state_dict)
-    iter_num = checkpoint["iter_num"]
-    best_val_loss = checkpoint["best_val_loss"]
+    if init_from == 'resume':
+        iter_num = checkpoint["iter_num"]
+        best_val_loss = checkpoint["best_val_loss"]
+
+if init_from == "lora_finetune":
+    out_dir = out_dir + "_lora_finetune"
+    os.makedirs(out_dir, exist_ok=True)
+    for p in model.parameters():
+        p.requires_grad = False
+    apply_lora(model, layer_types=lora_layer_types, rank=lora_rank, dropout=lora_dropout, alpha=lora_alpha)
```

Comment:
Looks like you apply LoRA to every Linear layer? Could you provide the ability to register lora for target modules (e.g. wq and wk only as suggested in the original paper)? The code may look like:

```python
def apply_lora(model, ..., target_modules=['wq', 'wk']:
    for name, layer in model.named_modules():
        if name.split(".")[-1] not in target_modules:
            continue
        # register lora parameterization

```

Reply:
I love this idea! I have a local implementation that does that, but I'll update this to follow suit.